Lib/asyncio/tasks.py

```diff
@@ -640,9 +640,34 @@ def __sleep0():
     yield
 
 
+def _check_delay(delay):
+    """Check if the value of 'delay' is valid."""
+
+    import sys
+    import math
+
+    SEC_TO_NS = 1000 * 1000 * 1000
+
+    if not isinstance(delay, (int, float)):
+        raise TypeError(f"{type(delay)} object cannot be interpreted as an integer")
+
+    # According to Modules/timemodule.c
+    if (delay > sys.maxsize / SEC_TO_NS):
+        raise OverflowError("timestamp too large to convert to C _PyTime_t")
+
+    if (delay < 0):
+        raise ValueError("sleep length must be non-negative")
+
+    if (math.isnan(delay)):
```

Comment:
Also, seems there need to add check that delay is finite number. Use something like `if not math.isfinite(delay): ...` for resolve that.

Reply:
This appears to be more accurate.

Thank you!